src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(man => (
+    century
+      ? man.sex === 'm' && Math.ceil(man.died / 100) === century
+      : man.sex === 'm'
+  ));
+  const menAges = men.map(man => man.died - man.born);
+  const averageMenAge = menAges.reduce((sum, age) => sum + age, 0) / men.length;
```

Comment:
you can do it by one iteration in reduce method
```suggestion
  const averageMenAge = menAges.reduce((sum, man) => sum + man.died - man.born, 0) / men.length;
```

Reply:
fixed!